wordpress-mobile/gutenberg-mobile
# Bridge: Harbor all RN deps and ability to get build via jitpack

This PR is part of the integration with wpandroid effort (https://github.com/wordpress-mobile/gutenberg-mobile/issues/163).

It introduces a new android package to harbor all the RN related dependencies instead of having them exposed in wpandroid. It also enables the ability to build the bridge project via jitpack so wpandroid can use it as a binary dependency.

Changes in this PR:

1. Jitpack config to jump in the bridge subfolder and build the android project. The artifact of this is then available as `com.github.wordpress-mobile:gutenberg-mobile:<commit_hash>`. wpandroid can use this to include the bridge. Notice that the bridge's module name (`react-native.gutenberg-bridge`) is unfortunately not published.
2. Use the unpkg.com exposed RN (and JavaScriptCore) When the `buildGutenbergFromSource` flag is set
3. When flag is false (also happens in jitpack build), the bridge module builds against binary dependencies for `react-native-svg`, `react-native-aztec` and `react-native-recyclerview-list`
4. Adds gradle wrapper to the bridge module for easier build on jitpack
5. `WPAndroidGlue` package to harbor the RN related init. The name is so poor, better we change it soon. Anyway, this package handles the initialization/teardown of the ReactNative root view, communicates with the bridge.

## To test

### Test A

0. On a fresh clone of the repo, without any submodule updates or yarn installs
1. Change into `react-native-gutenberg-bridge/android`
2. Run `./gradlew assemble`. It should succeed.
3. Notice in the backsroll the `Will use the unpkg.com exposed maven repo at https://unpkg.com/react-native@0.57.5/android` build log which indicates that the RN deps will be pulled from npmjs.com directly (so, no need for yarn install).

### Test B

0. On a normally set up local clone (submodules updated, yarn install, etc)
1. Build and run the RN app. It should run fine.
2. There should be no `Will use the unpkg.com exposed maven repo at https://unpkg.com/react-native@0.57.5/android` log from the build, which means RN was picked up from node_modules as normal.

### Test C

More testing could be done on the wpandroid side and using the `gb/single-gb-dep` branch there but, a wpandroid-side PR is in the works anyway. Feel free to do any testing though if curious.
